rust/types/src/metadata.rs

```diff
@@ -360,15 +371,35 @@ pub enum WhereClauseComparator {
 }
 
 #[derive(Clone, Debug, PartialEq)]
-pub enum WhereClauseListOperator {
+pub enum SetOperator {
     In,
     NotIn,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum MetadataSetValue {
+    Bool(Vec<bool>),
+    Int(Vec<i64>),
+    Float(Vec<f64>),
+    Str(Vec<String>),
+}
+
+impl MetadataSetValue {
+    pub fn into_vec(&self) -> Vec<MetadataValue> {
```

Comment:
can we do this by taking ownership and without copying

Reply:
For my use case, I want to construct a new `impl Iter<MetadataValue>` from `&MetadataSetValue`, where each individual `MetadataValue` will be used for further evaluation. Thus I cannot take ownership of the `self`.

The reason why I have a `&MetadataSetValue` and not a `MetadataSetValue` is because I hope to evaluate a `&Where` instead of a `Where`, as the same `&Where` need to be evaluated on both the log and the compacted data.

Reply:
Update: this is inlined.